invoices/invoiceregistry.go

```diff
@@ -880,15 +841,27 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 
 	switch invoiceHtlc.State {
 	case channeldb.HtlcStateCanceled:
-		return NewFailureResolution(
-			circuitKey, acceptHeight, result,
-		), nil
+		failureResolution := resolution.(*HtlcFailResolution)
+		failureResolution.AcceptHeight = acceptHeight
+
+		debugLog(fmt.Sprintf("failure resolution result "+
+			"outcome: %v, at accept height: %v",
+			failureResolution.Outcome, acceptHeight))
+
+		return failureResolution, nil
 
 	case channeldb.HtlcStateSettled:
+		failureResolution := resolution.(*HtlcSettleResolution)
+		failureResolution.AcceptHeight = acceptHeight
+
+		debugLog(fmt.Sprintf("settle resolution result "+
+			"outcome: %v, at accept height: %v",
+			failureResolution.Outcome, acceptHeight))
+
 		// Also settle any previously accepted htlcs. The invoice state
 		// is leading. If an htlc is marked as settled, we should follow
 		// now and settle the htlc with our peer.
-		for key, htlc := range invoice.Htlcs {
+		for _, htlc := range invoice.Htlcs {
```

Comment:
Just realized we're going to need to set the key on each resolution we notify. Otherwise they will all notify with the key of the htlc we just updated. 

Also need to rename `failureResolution`-> `settleResolution`